src/jackdaw/test/transports/kafka.clj

```diff
@@ -156,30 +158,32 @@
    injecting test messages"
   ([kafka-config topic-config serializers]
    (let [producer       (kafka/producer kafka-config byte-array-serde)
-         messages       (async/chan 1 (map (fn [x]
-                                             (log/info "producing: " x)
-                                             (try
-                                               (-> (apply-serializers serializers x)
-                                                   (build-record))
-                                               (catch Exception e
-                                                 (log/error e "kafka producer serialization error")
-                                                 (assoc x
-                                                        :serialization-error e))))))]
+         messages       (s/stream 1 (map (fn [x]
+                                           (log/info "producing: " x)
+                                           (try
+                                             (-> (apply-serializers serializers x)
+                                                 (build-record))
+                                             (catch Exception e
+                                               (log/error e "kafka producer serialization error")
+                                               (assoc x
+                                                      :serialization-error e))))))]
 
      (log/infof "started kafka producer: %s"
                 (select-keys kafka-config ["bootstrap.servers" "group.id"]))
 
-     (async/go-loop [{:keys [producer-record ack serialization-error] :as m} (async/<! messages)]
-       (cond
-         producer-record       (do (kafka/send! producer producer-record (deliver-ack ack))
-                                   (recur (async/<! messages)))
-         serialization-error   (do (deliver ack {:error :serialization-error
-                                                 :message (.getMessage serialization-error)})
-                                   (recur (async/<! messages)))
-         :else (do
-                 (.close producer)
-                 (log/infof "stopped kafka producer: "
-                            (select-keys kafka-config ["bootstrap.servers" "group.id"])))))
+     (d/loop [message (s/take! messages)]
+       (d/chain (d/future message)
```

Comment:
@cddr  `message` here is already a manifold deferred, why do you need to wrap it into a `future`?